commons/src/main/java/org/infinispan/commons/api/CacheContainerAdmin.java

```diff
@@ -0,0 +1,84 @@
+package org.infinispan.commons.api;
+
+import java.util.EnumSet;
+
+/**
+ * Administrative cache container operations.
+ *
+ * @author Tristan Tarrant
+ * @since 9.2
+ */
+
+public interface CacheContainerAdmin<C extends CacheContainerAdmin> {
```

Comment:
We need async APIs too for Vert.x and many other reactive use cases.

Reply:
https://issues.jboss.org/browse/ISPN-8526